packages/plugin-betswirl/src/actions/dice.ts

```diff
@@ -0,0 +1,279 @@
+import { z } from "zod";
+import { type Hex } from "viem";
+import {
+    type Action,
+    type IAgentRuntime,
+    type Memory,
+    type HandlerCallback,
+    type State,
+    elizaLogger,
+    type ActionExample,
+    composeContext,
+    generateObject,
+    ModelClass,
+} from "@moxie-protocol/core";
+import { MoxieWalletClient } from "@moxie-protocol/moxie-agent-lib/src/wallet";
+import {
+    CASINO_GAME_TYPE,
+    Dice,
+    DiceNumber,
+    MIN_SELECTABLE_DICE_NUMBER,
+    MAX_SELECTABLE_DICE_NUMBER,
+    slugById,
+    formatTxnUrl,
+} from "@betswirl/sdk-core";
+import { casinoBetParams, getMaxBetCountParam } from "../types";
+import {
+    getChainIdFromWallet,
+    getBetToken,
+    placeBet,
+    getBet,
+    getBetAmountInWei,
+} from "../utils/betswirl";
+import { formatTokenForMoxieTerminal } from "../utils/moxie";
+
+export const DiceBetParameters = z.object({
+    number: z
+        .number()
+        .gte(MIN_SELECTABLE_DICE_NUMBER)
+        .lte(MAX_SELECTABLE_DICE_NUMBER)
+        .describe("The number to bet on"),
+    ...casinoBetParams,
+    ...getMaxBetCountParam(CASINO_GAME_TYPE.DICE),
+});
+export const diceTemplate = `
+Extract the following details to play on Dice:
+- **betAmount** (String?): The amount to wager.
+- **number** (Number?): The number to bet on. Can be from ${MIN_SELECTABLE_DICE_NUMBER} to ${MAX_SELECTABLE_DICE_NUMBER}.
+- **token** (String?): The token symbol.
+Where "?" indicates that the value is optional.
+
+Provide the values in the following JSON format:
+\`\`\`json
+{
+    "betAmount": string?,
+    "number": number?,
+    "token": string?
+}
+\`\`\`
+
+Here are example messages and their corresponding responses:
+
+**Message 1**
+\`\`\`
+Bet 0.01 ETH above 44
+\`\`\`
+
+**Response 1**
+\`\`\`json
+{
+    "betAmount": "0.01",
+    "number": 44,
+    "token" "ETH"
+}
+\`\`\`
+
+**Message 2**
+\`\`\`
+Roll the dice with 0.01 ETH on 23
+\`\`\`
+
+**Response 2**
+\`\`\`json
+{
+    "betAmount": "0.5",
+    "number": 23,
+    "token": "ETH",
+}
+\`\`\`
+
+** Message 3 **
+\`\`\`
+Roll a dice for me
+\`\`\`
+
+** Response 3 **
+\`\`\`json
+{
+    "betAmount": null,
+    "number": null,
+    "token": null,
+}
+\`\`\`
+
+** Message 4 **
+\`\`\`
+I want to bet on the 8
+\`\`\`
+
+** Response 4 **
+\`\`\`json
+{
+    "betAmount": null,
+    "number": 8,
+    "token": null,
+}
+\`\`\`
+
+** Message 5 **
+\`\`\`
+I want to bet 0.01 on 8
+\`\`\`
+
+** Response 5 **
+\`\`\`json
+{
+    "betAmount": "0.01",
+    "number": 8,
+    "token": null,
+}
+\`\`\`
+
+** Message 6 **
+\`\`\`
+I want to bet my ETH on 8
+\`\`\`
+
+** Response 6 **
+\`\`\`json
+{
+    "betAmount": null,
+    "number": 8,
+    "token": "ETH",
+}
+\`\`\`json
+
+Here are the recent user messages for context:
+{{recentMessages}}
+`;
+export const diceAction: Action = {
+    name: "DICE",
+    similes: ["ROLL_A_DICE", "DICE_ROLL", "BETSWIRL_DICE"],
```

Comment:
Also add more similes to help to train agent

recommended to generate w/ AI if you want to make things faster and easier if you'd like